src/modules/sdlog2/sdlog2.c

```diff
@@ -584,6 +596,12 @@ static void *logwriter_thread(void *arg)
 		if (++poll_count == 10) {
 			fsync(log_fd);
 			poll_count = 0;
+
+			/* check if space is available, if not stop everything */
+			if (check_free_space() != OK) {
```

Comment:
We know how much we're writing (look at the increasing total write counter). There are other processes writing as well, but you could check every 20m. That seems the proper way to throttle the check.

I would also suggest a hysteresis value of ~40-100M. The microSD cards we use are huge, and sdlog2 is not the only process writing.